modelling/src/main/java/org/axonframework/modelling/command/StatefulCommandHandlingComponent.java

```diff
@@ -0,0 +1,308 @@
+    private boolean loadModelsEagerly = false;
+
+    /**
+     * Initializes a new stateful command handling component with the given {@code name}.
+     * @param name The name of the component
+     */
+    private StatefulCommandHandlingComponent(String name) {
+        this.name = name;
+        this.handlingComponent = SimpleCommandHandlingComponent.forComponent(name);
+    }
+
+    /**
+     * Creates a new stateful command handling component with the given {@code name}.
+     * @param name The name of the component
+     * @return A new stateful command handling component with the given {@code name}
+     */
+    public static StatefulCommandHandlingComponent forName(String name) {
+        return new StatefulCommandHandlingComponent(name);
+    }
+
+    /**
+     * Tells this component to load models eagerly. This means that all models are loaded when the command is handled
+     * and the resolved identifier is non-null.
+     *
+     * @return this component for fluent interfacing
+     */
+    public StatefulCommandHandlingComponent loadModelsEagerly() {
+        loadModelsEagerly = true;
+        return this;
+    }
+
+
+    @Override
+    public StatefulCommandHandlingComponent subscribe(
+            @Nonnull QualifiedName name,
+            @Nonnull StatefulCommandHandler commandHandler
+    ) {
+        handlingComponent.subscribe(name, ((command, context) -> {
+            try {
+                var models = new StatefulCommandModelModelContainer(
+                        command,
+                        context,
+                        loadModelsEagerly
+                );
+                return commandHandler.handle(command, models, context);
+            } catch (Exception e) {
+                return MessageStream.failed(e);
+            }
+        }));
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageStream.Single<? extends CommandResultMessage<?>> handle(@Nonnull CommandMessage<?> command,
+                                                                   @Nonnull ProcessingContext context) {
+        return handlingComponent.handle(command, context);
+    }
+
+    @Override
+    public StatefulCommandHandlingComponent subscribe(@Nonnull QualifiedName name,
+                                                      @Nonnull CommandHandler commandHandler) {
+        handlingComponent.subscribe(name, commandHandler);
+        return this;
+    }
+
+    @Override
+    public Set<QualifiedName> supportedCommands() {
+        return handlingComponent.supportedCommands();
+    }
+
+    @Override
+    public <ID, M> StatefulCommandHandlingComponent registerModel(
+            String name,
+            Class<M> modelClass,
+            Function<CommandMessage<?>, ID> commandIdResolver,
+            ModelLoader<ID, M> loadFunction) {
+        if (modelDefinitions.stream().anyMatch(md -> md.name().equals(name) && md.modelClass().equals(modelClass))) {
```

Comment:
So can I have same model names for different types? What is the use case for that?
If yes please adjust the exception message.